### DISCUSSION TITLE
Error while starting zigbee-herdsman

### DISCUSSION THREAD

**Discussion by mtthsdzwn** [Koenkk/zigbee2mqtt#6443] [General]
I'm trying to set up Home Assistant on my Raspberry Pi 4. I don't want so dedicate my rpi fully to Home Assistant, so I decided to use the docker image, which is working fine. I have a [zzh CC2652 stick](https://electrolama.com/projects/zig-a-zig-ah/) plugged in and flashed, this also seems to be working fine. The zzh is recognised and installed by Home Assistant.  

I found out that the docker-based HA does not include the supervisor to install the zigbee2mqtt add-on, but I understand that I can use the zigbee2mqtt docker image, which is what I did. However, zigbee2mqtt does not start up. From `docker logs zigbee2mqtt`, I see (among many other things)
```
Zigbee2MQTT:info  2021-02-24 21:55:17: Logging to console and directory: '/app/data/log/2021-02-24.21-55-16' filename: log.txt
Zigbee2MQTT:info  2021-02-24 21:55:17: Starting Zigbee2MQTT version 1.17.1 (commit #99274c7)
Zigbee2MQTT:info  2021-02-24 21:55:17: Starting zigbee-herdsman (0.13.59)
Zigbee2MQTT:error 2021-02-24 21:55:17: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-02-24 21:55:17: Failed to start zigbee
Zigbee2MQTT:error 2021-02-24 21:55:17: Exiting...
Zigbee2MQTT:error 2021-02-24 21:55:17: Error: Error while opening serialport 'Error: Error: No such file or directory, cannot open /dev/ttyACM0'
    at Znp.<anonymous> (/app/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:136:32)
```

The zzh is at `/dev/ttyUSB0`, not `ttyACM0` as in the error message. I have referenced this correctly (I think) in the [docker-compose.yml](https://gist.github.com/mtthsdzwn/e659f74b392e0100544aa8943963ae18#file-docker-compose-yml). (I also use this docker-compose file to install some other applications that I edited out. That is why I am using version 2.1.)

I'll gladly provide more information if needed, but then you need to tell me what you need...

I am new to all of this, so I don't really know where to start looking for an answer. I also would appreciate a not-overly-technical explanation of what is going wrong. If this is not the best place to look for an answer, could you please tell me where to ask my question?

Thanks!
Matthijs

**Comment by m0wlheld** [comment]
Add 

    volumes:
      - /dev/ttyUSB0:/dev/ttyACM0

to the service definition of Zigbee2mqtt in your compose file. 

OR set the correct serial path in the configuration.yaml file of Zigbee2mqtt.

**Reply by mtthsdzwn** [reply]
Thanks, works! 👍

**Comment by 063405** [comment]
Hello,
I have the same problem, getting a similar message.
`Zigbee2MQTT:error 2021-03-13 10:16:58: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-03-13 10:16:58: Failed to start zigbee
Zigbee2MQTT:error 2021-03-13 10:16:58: Exiting...
Zigbee2MQTT:error 2021-03-13 10:16:58: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
`
I guess the serial path is correct as it is correct in the configuration.yaml

Any guesses, please? I did the installations more than 10 times, so I am pretty desperate. 

(Also note that my knowledge in programming is almost zero, I am just reading a lot and following guides, as I want to setup a home automation for fun)

Thanks, I appreciate any help
